cmd/git/git.go

```diff
@@ -1026,167 +1028,76 @@ func (j *DSGit) SetParentCommitFlag(richItem map[string]interface{}) (err error)
-		locAdded, _ := doc["lines_added"].(int64)
-		locRemoved, _ := doc["lines_removed"].(int64)
-		locChanged, _ := doc["lines_changed"].(int64)
-		_, squashedCommit := j.OrphanedMap["sha"]
-		if sMessage != "" {
-			message = &sMessage
-		}
-		if sTitle != "" {
-			title = &sTitle
-		}
-		parents, _ := doc["parents"].([]string)
-		authoredInTz, _ := doc["author_date"].(time.Time)
+		commit.URL, _ = doc["commit_url"].(string)
+		commit.SHA, _ = doc["hash"].(string)
+		commit.ShortHash, _ = doc["hash_short"].(string)
+		commit.Source = doc["commit_repo_type"].(string)
+		commit.Message, _ = doc["message"].(string)
+		_, commit.MergeCommit = j.OrphanedMap["sha"]
+		commit.ParentSHAs, _ = doc["parents"].([]string)
+		commit.AuthoredTimestamp, _ = doc["author_date"].(time.Time)
 		authoredDt, _ := doc["utc_author"].(time.Time)
-		authoredTz, _ := doc["tz"].(float64)
-		committedInTz, _ := doc["commit_date"].(time.Time)
-		committedDt, _ := doc["utc_commit"].(time.Time)
-		committedTz, _ := doc["commit_tz"].(float64)
+		commit.CommittedTimestamp, _ = doc["commit_date"].(time.Time)
 		createdOn := authoredDt
-		commitRoles := []*models.CommitRole{}
+		commit.Orphaned = doc["orphaned"].(bool)
+		commit.SyncTimestamp = time.Now()
+		commitRoles := []insights.Contributor{}
 		identsAry, okIdents := doc["idents"].([][3]string)
 		identTypesAry, okIdentTypes := doc["ident_types"].([]string)
 		if okIdents && okIdentTypes {
 			for i := range identTypesAry {
-				var (
-					dt     time.Time
-					dtInTz time.Time
-					tz     float64
-				)
+				commitRole := insights.Contributor{}
 				ident := identsAry[i]
 				identType := identTypesAry[i]
-				if identType == "committer" {
-					dt = committedDt
-					dtInTz = committedInTz
-					tz = committedTz
-				} else {
-					dt = authoredDt
-					dtInTz = authoredInTz
-					tz = authoredTz
-				}
+				commitRole.Role = identType
+				commitRole.Weight = 1.0
 				name := ident[0]
 				username := ""
 				email := ident[2]
 				name, username = shared.PostprocessNameUsername(name, username, email)
-				userUUID := shared.UUIDAffs(ctx, source, email, name, username)
-				identity := &models.Identity{
-					ID:           userUUID,
-					DataSourceID: source,
-					Name:         name,
-					Username:     username,
+				nameList := strings.Split(name, "")
+				commitRole.Identity = user.UserIdentity{
+					Email:      email,
+					FirstName:  nameList[0],
+					IsVerified: false,
+					Username:   username,
+					Source:     source.Slug,
 				}
-				commitRoleUUID := shared.UUIDNonEmpty(ctx, docUUID, fmt.Sprintf("%s-%d", identType, i), userUUID)
-				commitRole := &models.CommitRole{
-					ID:       commitRoleUUID,
-					Name:     identType,
-					User:     identity,
-					Weight:   1.0,
-					When:     strfmt.DateTime(dt),
-					WhenInTz: strfmt.DateTime(dtInTz),
-					WhenTz:   tz,
+				if len(nameList) > 1 {
+					commitRole.Identity.LastName = strings.Join(nameList[1:], " ")
```

Comment:
How can we know that last name is anything after the first word in the name?

Reply:
removed.